### PR TITLE
Fill in some of the peer syncing logic

### PR DESCRIPTION
**Pull request by ErisMik**

### CODE REVIEW

File: bookchain-miner/src/mainpeers.cpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        curl_easy_cleanup(curl);
+    }
+
+    return (returnString.empty()) ? "{}" : returnString;
+}
+
+void discoverInitialPeers(PeersList& peersList) {
+    const std::array<std::string, 1> peerHints = {"127.0.0.1:8000"};
+
+    auto discoverPeer = [&peersList](const std::string& peerHint) {
```
Comment (MuchToKnow) [inline bookchain-miner/src/mainpeers.cpp:75]: Can you explain the square bracket notation in this context for a c++ noob :)

**Review by MuchToKnow — APPROVED** [review]
The peering looks super solid so far and it's easy to understand :+1:  Also like to see the tests :)

Nice PR

File: bookchain-miner/src/mainpeers.cpp (hunk 1 above)
Comment (ErisMik) [reply bookchain-miner/src/mainpeers.cpp:75]: So this is a lambda as you've probably guessed. C++ doesn't capture variables from the outside scope into the lambda automatically for most cases. The square brackets are to explicitly capture variables and put them into the scope of the lambda. 

https://en.cppreference.com/w/cpp/language/lambda#Lambda_capture

**Comment by ErisMik** [comment]
> The peering looks super solid so far and it's easy to understand +1 Also like to see the tests :)
> 
> Nice PR

Thanks!

**Pull request merged**